app/components/calendar-input.js

```diff
@@ -86,13 +87,22 @@ export default Ember.TextField.extend({
 
   initializeConfig() {
     this.set("datePickerConfig", DEFAULT_PICKADATE_CONFIG);
+
     const enablePastDate = this.get("enablePastDate");
     if (enablePastDate) {
       this.set("datePickerConfig", {
         ...DEFAULT_PICKADATE_CONFIG,
         min: MIN_DATE
       });
     }
+
+    const formatSubmit = this.get("formatSubmit");
+    if (formatSubmit) {
+      this.set("datePickerConfig", {
+        ...DEFAULT_PICKADATE_CONFIG,
+        format: formatSubmit
+      });
+    }
   },
```

Comment:
There's no need to `set(...)` 3 times : ) 

```suggestion
  initializeConfig() {
    let config = { ...DEFAULT_PICKADATE_CONFIG }

    const enablePastDate = this.get("enablePastDate");
    if (enablePastDate) {
      config.min = MIN_DATE;
    }

    const formatSubmit = this.get("formatSubmit");
    if (formatSubmit) {
      config.format = formatSubmit;
    }

    this.set("datePickerConfig", config);
  },
```